components/automate-deployment/pkg/airgap/bundle_creator.go

```diff
@@ -438,61 +438,42 @@ func (creator *InstallBundleCreator) downloadPackageAndKeys(p habpkg.VersionedPa
 		return nil
 	}
 
-	return creator.hartCache.CacheArtifact(p, func(writer io.Writer) error {
-		header, err := creator.downloadPackageWithRetries(p, writer, depotClient, progress)
-		if err == nil {
-			if creator.keyCache.IsCached(header.KeyName) {
-				return nil // We don't log here since nearly every key is cached.
-			}
-
-			return creator.keyCache.CacheKey(header.KeyName, func(writer io.Writer) error {
-				return creator.downloadOriginKeyWithRetries(header.KeyName, writer, depotClient, progress)
-			})
-		}
-		return err
-	})
+	return creator.downloadWithRetries(p, depotClient, progress)
 }
 
-func (creator *InstallBundleCreator) downloadPackageWithRetries(p habpkg.VersionedPackage, writer io.Writer,
-	depotClient depot.Client, progress InstallBundleCreatorProgress) (*depot.HartHeader, error) {
-
+func (creator *InstallBundleCreator) downloadWithRetries(p habpkg.VersionedPackage, depotClient depot.Client, progress InstallBundleCreatorProgress) error {
 	name := habpkg.Ident(p)
-
-	var header *depot.HartHeader
-	err := creator.withRetry(func(tryCount int, ri retryInfo) error {
-		progress.Downloading(name, tryCount)
-		var err error
-		header, err = depotClient.DownloadPackage(p, writer)
-		if err != nil {
-			if ri.willRetry {
-				progress.RetriableDownloadError(name, err.Error(), ri.delay)
+	return creator.withRetry(func(tryCount int, ri retryInfo) error {
+		return creator.hartCache.CacheArtifact(p, func(writer io.Writer) error {
+			progress.Downloading(name, tryCount)
+			header, err := depotClient.DownloadPackage(p, writer)
+			if err != nil {
+				if ri.willRetry {
+					progress.RetriableDownloadError(name, err.Error(), ri.delay)
+				}
+				return err
 			}
-			return err
-		}
-		progress.DownloadComplete(name, false)
-		return nil
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	return header, nil
-}
-
-func (creator *InstallBundleCreator) downloadOriginKeyWithRetries(keyname depot.OriginKeyName, writer io.Writer,
-	depotClient depot.Client, progress InstallBundleCreatorProgress) error {
 
-	return creator.withRetry(func(tryCount int, ri retryInfo) error {
-		progress.Downloading(string(keyname), tryCount)
-		err := depotClient.DownloadOriginKey(keyname, writer)
-		if err != nil {
-			if ri.willRetry {
-				progress.RetriableDownloadError(string(keyname), err.Error(), ri.delay)
+			progress.DownloadComplete(name, false)
+			if creator.keyCache.IsCached(header.KeyName) {
+				return nil // We don't log here since nearly every key is cached.
 			}
-			return err
-		}
-		progress.DownloadComplete(string(keyname), false)
-		return nil
+
+			return creator.withRetry(func(tryCount int, ri retryInfo) error {
```

Comment:
I'm considering removing this inner retry.  I'm not sure which is more confusing, both downloads sharing the same retry count, or this key download potentially being retried tries*tries times.  I'm thinking sharing the same retry count is better.